core/include/detray/masks/masks.hpp

```diff
@@ -189,6 +189,12 @@ class mask {
     DETRAY_HOST_DEVICE
     auto volume_link() -> links_type& { return _volume_link; }
 
+    /// @returns the masks center in global cartesian coordinates
+    template <typename transform3_t>
+    DETRAY_HOST_DEVICE auto center(const transform3_t& trf) const {
+        return _shape.center(trf, _values);
```

Comment:
Is the following possible instead of having center function all mask types?
```c++
if constexpr (shape_t == annulus2D<>){
   return annulus.center()
}
return trf.translation()
```

Reply:
Sure, I can do that. The shapes are not only used in the masks though, meaning only the annulus and the unbound shape will have a `center()` method and the others not

Reply:
I changed it now, so that the `center()` method on the surface returns the same thing it returns in ACTS and the `centroid()` method gives us a point on the surface (also useful for some cylinders)